cmake/compiler.cmake

```diff
@@ -264,11 +256,35 @@ macro(enable_tnt_compile_flags)
             # example, having "returns-nonnull-attribute" may lead to fail in
             # the typeof(*obj) when obj is NULL, even though there is nothing
             # related to return.
-
-            set(SANITIZE_FLAGS "-fsanitize=${SANITIZE_FLAGS} -fno-sanitize-recover=${SANITIZE_FLAGS}")
-
-            add_compile_flags("C;CXX" "${SANITIZE_FLAGS}")
+            null
+            nonnull-attribute
+            nullability-arg
+            returns-nonnull-attribute
+            nullability-assign
+            nullability-return
+        )
+        # GCC has no "function" UB check. See details here:
+        # https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html#index-fsanitize_003dundefined
+        if(NOT CMAKE_C_COMPILER_ID STREQUAL "GNU")
```

Comment:
Do we need this check? Isn't [this](https://github.com/tarantool/tarantool/pull/10605/commits/fba798fbcccff96726aefe3a4781fdc576567d72#diff-abe123d6b477234b7b16bf1b8d33438cc8e62bafaed6e92b288f319aca3adaceR227) enough?

I suppose it will be the proper check when we enabling the UBSan for GCC, but this shouldn't be the part of this patchset.

Reply:
The condition has been removed, I'll add it in the patch with GCC/UBSan support.

```diff
diff --git a/cmake/compiler.cmake b/cmake/compiler.cmake
index 4d87d14747..c2ea1614e3 100644
--- a/cmake/compiler.cmake
+++ b/cmake/compiler.cmake
@@ -262,28 +262,21 @@ macro(enable_tnt_compile_flags)
             returns-nonnull-attribute
             nullability-assign
             nullability-return
+            # Not interested in function type mismatch errors.
+            function
         )
-        # GCC has no "function" UB check. See details here:
-        # https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html#index-fsanitize_003dundefined
-        if(NOT CMAKE_C_COMPILER_ID STREQUAL "GNU")
-            string(JOIN "," UBSAN_IGNORE_OPTIONS
-                ${UBSAN_IGNORE_OPTIONS}
-                # Not interested in function type mismatch errors.
-                function
-            )
-            # XXX: To get nicer stack traces in error messages.
-            set(SANITIZE_FLAGS "${SANITIZE_FLAGS} -fno-omit-frame-pointer")
-            # Enable UndefinedBehaviorSanitizer support.
-            # This flag enables all supported options (the documentation
-            # on site is not correct about that moment, unfortunately)
-            # except float-divide-by-zero. Floating point division by zero
-            # behaviour is defined without -ffast-math and uses the
-            # IEEE 754 standard on which all NaN tagging is based.
-            set(SANITIZE_FLAGS "${SANITIZE_FLAGS} -fsanitize=undefined")
-            set(SANITIZE_FLAGS "${SANITIZE_FLAGS} -fno-sanitize=${UBSAN_IGNORE_OPTIONS}")
-            # Print a verbose error report and exit the program.
-            set(SANITIZE_FLAGS "${SANITIZE_FLAGS} -fsanitize-recover=undefined")
-        endif()
+        # XXX: To get nicer stack traces in error messages.
+        set(SANITIZE_FLAGS "${SANITIZE_FLAGS} -fno-omit-frame-pointer")
+        # Enable UndefinedBehaviorSanitizer support.
+        # This flag enables all supported options (the documentation
+        # on site is not correct about that moment, unfortunately)
+        # except float-divide-by-zero. Floating point division by zero
+        # behaviour is defined without -ffast-math and uses the
+        # IEEE 754 standard on which all NaN tagging is based.
+        set(SANITIZE_FLAGS "${SANITIZE_FLAGS} -fsanitize=undefined")
+        set(SANITIZE_FLAGS "${SANITIZE_FLAGS} -fno-sanitize=${UBSAN_IGNORE_OPTIONS}")
+        # Print a verbose error report and exit the program.
+        set(SANITIZE_FLAGS "${SANITIZE_FLAGS} -fsanitize-recover=undefined")
         add_compile_flags("C;CXX" "${SANITIZE_FLAGS}")
     endif()

```

Reply:
I've removed relevant patch, I'll add it to another PR.